cirq-google/cirq_google/engine/qcs_notebook.py

```diff
@@ -152,3 +134,68 @@ def get_qcs_objects_for_notebook(
         processor_id=processor_id,
         is_simulator=is_simulator,
     )
+
+
+def get_hardware_engine_and_authenticate_user(
+    project_id: Optional[str] = None, clear_output: bool = False
+) -> AbstractEngine:
+    """Authenticates on Google Cloud and returns an Engine for running against hardware.
+
+    Args:
+        project_id: Optional explicit Google Cloud project id. Otherwise,
+            this defaults to the environment variable GOOGLE_CLOUD_PROJECT.
+            By using an environment variable, you can avoid hard-coding
+            personal project IDs in shared code.
+
+        clear_output: Optional bool for whether to clear output before
+            authenticating. Defaults to false.
+
+    Returns:
+        An instance of AbstractEngine.
+
+    Raises:
+        Exception: if authentication fails.
+    """
+
+    try:
+        _authenticate_user(clear_output)
+        return get_engine(project_id)
+    except Exception as exc:
+        print(
+            "Authentication failed, you may not have permission to access a"
```

Comment:
Should the "authentication failed" message occur en the authentication step? e.g., what happens if `get_engine` raises an exception?

Reply:
n/a since this api was removed